src/ui/connectionmanager.cc

```diff
@@ -0,0 +1,190 @@
+      connection_dialog_, &QDialog::show);
+  connect(connection_dialog_, &QDialog::accepted,
+      this, &ConnectionManager::connectionDialogAccepted);
+  connect(kill_locally_created_server_action_, &QAction::triggered,
+      this, &ConnectionManager::killLocalServer);
+  connect(disconnect_action_, &QAction::triggered,
+        this, &ConnectionManager::disconnect);
+}
+
+ConnectionManager::~ConnectionManager() {
+  connection_dialog_->deleteLater();
+}
+
+QAction* ConnectionManager::showConnectionDialogAction() {
+  return show_connection_dialog_action_;
+}
+
+QAction* ConnectionManager::disconnectAction() {
+  return disconnect_action_;
+}
+
+QAction* ConnectionManager::killLocallyCreatedServerAction() {
+  return kill_locally_created_server_action_;
+}
+
+void ConnectionManager::locallyCreatedServerStarted() {
+  QTextStream out(veles::ui::LogWidget::output());
+  out << "Process of locally created server started." << endl;
+  kill_locally_created_server_action_->setEnabled(true);
+}
+
+void ConnectionManager::locallyCreatedServerFinished(int exit_code,
+    QProcess::ExitStatus exit_status) {
+  QTextStream out(veles::ui::LogWidget::output());
+  out << "Process of locally created server finished." << endl;
+  kill_locally_created_server_action_->setEnabled(false);
+  server_process_->deleteLater();
+  server_process_ = nullptr;
+}
+
+void ConnectionManager::connectionDialogAccepted() {
+  if(connection_dialog_->runANewServer()) {
+    startLocalServer();
+  }
+
+  // TODO actually connect with the server
+  // Can be done once C++ client library is created.
+}
+
+void ConnectionManager::startLocalServer() {
+  if(server_process_) {
+    server_process_->deleteLater();
+  }
+
+  server_process_ = new QProcess(this);
+  server_process_->setProcessChannelMode(QProcess::MergedChannels);
+  connect(server_process_, &QProcess::started,
+      this, &ConnectionManager::locallyCreatedServerStarted);
+  connect(server_process_, static_cast<void(QProcess::*)
+      (int, QProcess::ExitStatus)>(&QProcess::finished),
+      this, &ConnectionManager::locallyCreatedServerFinished);
+  connect(server_process_, &QIODevice::readyRead,
+        this, &ConnectionManager::serverProcessReadyRead);
+
+  QString server_file_path = connection_dialog_->serverScript();
+  QFileInfo server_file(server_file_path);
+  QString directory_path = server_file.absoluteDir().path();
+  QString server_file_name = server_file.fileName();
+
+  server_process_->setWorkingDirectory(directory_path);
+  QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
+  env.insert("PYTHONUNBUFFERED", "1");
+  server_process_->setProcessEnvironment(env);
+
+  QStringList arguments;
+  arguments
+      << server_file_name
+      << connection_dialog_->databaseFile()
+      << QString("%1:%2").arg(connection_dialog_->serverHost())
+      .arg(connection_dialog_->serverPort());
+
+  QString python_interpreter_executable("python3");
```

Comment:
By default when installing Python on Windows it doesn't create python2/3 executables - on this platform you should instead use py.exe which checks in Windows registry where interpreter for requested Python version is located.

However I'm thinking about how we should handle starting the server in the long run - current approach launches it in global Python env which requires user to install our requirements in global Python env - which most likely is not desirable.